Libs/MRML/Core/vtkITKTransformConverter.h

```diff
@@ -164,12 +180,90 @@ void vtkITKTransformConverter::RegisterInverseTransformTypes()
   itk::TransformFactory<InverseThinPlateSplineTransformFloatType>::RegisterTransform();
   itk::TransformFactory<InverseThinPlateSplineTransformDoubleType>::RegisterTransform();
 
+  itk::TransformFactory<itk::InverseThinPlateSplineKernelTransform<float, 2>>::RegisterTransform();
+  itk::TransformFactory<itk::InverseThinPlateSplineKernelTransform<double, 2>>::RegisterTransform();
+
   typedef itk::ThinPlateSplineKernelTransform<float, 3> ThinPlateSplineTransformFloatType;
   typedef itk::ThinPlateSplineKernelTransform<double, 3> ThinPlateSplineTransformDoubleType;
 
   // by default they are not registered
   itk::TransformFactory<ThinPlateSplineTransformFloatType>::RegisterTransform();
   itk::TransformFactory<ThinPlateSplineTransformDoubleType>::RegisterTransform();
+
+  itk::TransformFactory<itk::ThinPlateSplineKernelTransform<float, 2>>::RegisterTransform();
+  itk::TransformFactory<itk::ThinPlateSplineKernelTransform<double, 2>>::RegisterTransform();
+}
+
+//----------------------------------------------------------------------------
+template <typename T>
+itk::Matrix<T, 3, 3> vtkITKTransformConverter::Matrix2Dto3D(itk::Matrix<T, 2, 2> m2D)
+{
+  itk::Matrix<T, 3, 3> m3D;
+  m3D.SetIdentity();
+  for (unsigned int i = 0; i < 2; i++)
+  {
+    for (unsigned int j = 0; j < 2; j++)
+    {
+      m3D[i][j] = m2D(i, j);
+    }
+  }
+
+  // compute scale for Z direction
+  itk::Vector<T, VTKDimension> directionCosine{};
+  directionCosine[0] = m3D[0][0];
+  directionCosine[1] = m3D[0][1];
+  auto scaleX = directionCosine.GetNorm();
+  directionCosine[0] = m3D[1][0];
+  directionCosine[1] = m3D[1][1];
+  auto scaleY = directionCosine.GetNorm();
+  T scaleZ = std::sqrt(scaleX * scaleY); // geometric mean
+  m3D(2, 2) = scaleZ;
```

Comment:
We can simplify this to something like this:

    m3D(2,2) = (vnl_determinant(m2D) < 0.0) ? -scaleZ : scaleZ;

and we don't need the NDEBUG part